pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java

```diff
@@ -253,12 +252,29 @@ public int getNumKeys() {
   }
 
   private class ArrayBasedHolder implements RawKeyHolder {
-    // TODO: using bitmap might better
```

Comment:
Should we consider using `BitSet` to replace this boolean array?

Reply:
No it’s not a good idea because it creates data dependencies between any contiguous set of 64 groups as they all need to update the same word. I tried it and it was about 30% worse.